conformance/utils/kubernetes/portset.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// PortStack is a wrapper for handling sets of ports.
+type PortStack []v1beta1.PortNumber
+
+func (s *PortStack) PopPort() (v1beta1.PortNumber, bool) {
```

Comment:
could use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set ?

Reply:
See https://github.com/kubernetes-sigs/gateway-api/pull/1217#discussion_r1136225015 :)

Reply:
ha! nice feel free to ignore 👍🏽